cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -551,4 +551,25 @@ public static String getJavaSystemPropertiesToString(List<SystemProperty> javaSy
         }
         return String.join(" ", javaSystemPropertiesList);
     }
+
+    /**
+     * This method transforms a String into a List of Strings, where entry = line from input.
+     * The lines beginning with '#' (comments) are ignored.
+     * @param config ConfigMap data as a String
+     * @return List of String key=value
+     */
+    public static List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> validLines = new ArrayList<>();
+        if (config != null) {
+            List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+
+            for (String line : allLines) {
+                if (!line.startsWith("#") && !line.isEmpty()) {
```

Comment:
I wonder if it is possible that a line might be ` #...` e.g. a whitespace at the beginning of a line, and if we should catch this and count this as a comment line.
Also could a comment be appended to the end of an input line?

Reply:
What comment do you mind?